master/internal/api_experiment.go

```diff
@@ -1004,12 +1004,19 @@ func (a *apiServer) PatchExperiment(
 	}
 
 	if req.Experiment.Labels != nil {
-		var reqLabelList []string
+		// avoid duplicate keys
+		reqLabelSet := make(map[string]struct{}, len(req.Experiment.Labels.Values))
```

Comment:
this is more of a nitpick, but I think you could use `map[string]bool` and setting to `true` instead of making structs

Reply:
if it's 0 bytes then AOK